src/plugins/data_source/server/routes/test_connection.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { schema } from '@osd/config-schema';
+import { IRouter, OpenSearchClient } from 'opensearch-dashboards/server';
+import { DataSourceAttributes } from '../../common/data_sources';
+import { DataSourceConnectionValidator } from './data_source_connection_validator';
+import { DataSourceServiceSetup } from '../data_source_service';
+import { CryptographyServiceSetup } from '../cryptography_service';
+
+export const registerTestConnectionRoute = (
+  router: IRouter,
+  dataSourceServiceSetup: DataSourceServiceSetup,
+  cryptography: CryptographyServiceSetup
+) => {
+  router.post(
+    {
+      path: '/internal/data-source-management/validate',
+      validate: {
+        body: schema.object({
+          id: schema.string(),
+          endpoint: schema.string(),
+          auth: schema.maybe(
+            schema.object({
+              type: schema.oneOf([schema.literal('username_password'), schema.literal('no_auth')]),
+              credentials: schema.oneOf([
+                schema.object({
+                  username: schema.string(),
+                  password: schema.string(),
+                }),
+                schema.literal(null),
+              ]),
+            })
+          ),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      const dataSource: DataSourceAttributes = request.body as DataSourceAttributes;
+
+      const dataSourceClient: OpenSearchClient = await dataSourceServiceSetup.getTestingClient(
+        {
+          dataSourceId: dataSource.id || '',
+          savedObjects: context.core.savedObjects.client,
+          cryptography,
+        },
+        dataSource
+      );
+
+      try {
+        const dsValidator = new DataSourceConnectionValidator(dataSourceClient);
+
+        await dsValidator.validate();
```

Comment:
can we close the client using `client.close()` at the end of the call? Otherwise, if user click the button multiple times, and each time it creates a new client,  it could fill up the sockets or memory